packages/ui/components/Icon/Icon.tsx

```diff
@@ -1,37 +1,44 @@
-import classnames from 'classnames';
 import React, { PropsWithRef } from 'react';
+import styled, { css } from 'styled-components';
 
 import { icons } from './icons';
 
-
 export type IconType = keyof typeof icons;
+export enum IconSize {
+  small = 'small',
+  medium = 'medium',
+  large = 'large'
+}
+export type IconColors = 'primary' | 'secondary' | 'tertiary';
 
 export interface IconProps extends PropsWithRef<any> {
   icon: IconType;
-  color?: string;
-  size?: 'large' | 'small' | 'medium' | number;
+  size?: IconSize;
+  color?: IconColors | string;
+  stroke: string;
+  strokeWidth: string;
 }
 
-export const sizeMap = {
-  small: '4',
-  medium: '10',
-  large: '20'
-};
-
-
-export const Icon: React.FunctionComponent<IconProps> = ({
+export const Icon = styled<React.FC<IconProps>>(({
   icon,
   color,
-  className,
-  size = 'medium',
+  stroke,
+  strokeWidth,
   ...props
 }) => {
-  const colorClass = color ? `text-${color}` : null;
-
-  const sizeClass = `h-${typeof size === 'number' ? size : sizeMap[size]}`;
-
-  const klass = classnames('icon', className, sizeClass, colorClass);
   const Ikon = icons[icon];
   if (!Ikon) return null;
-  return <Ikon {...props} className={klass} />;
-};
+  return <Ikon {...props} />;
+})`
+  height:${({ size }) => (size === IconSize.small ? '15px' : size === IconSize.large ? '45px' : '30px')};
```

Comment:
changed to:
`height: ${({ size, theme }) => size ? theme.iconSizes[size] : theme.iconSizes['medium']};`

Reply:
Allow size as customized number as well since there are different size needs, e.g.` <Icon size={75}>`
```
export interface IconProps {
...
size?: keyof typeof theme.iconSizes | number;
...
}

```
`height: ${({ size, theme }) => !size ? theme.iconSizes['medium'] : typeof size === 'number' ? `${size}px` : theme.iconSizes[size]};`